webapps-schema/src/main/java/io/camunda/webapps/schema/entities/usermanagement/EntityJoinRelation.java

```diff
@@ -7,40 +7,43 @@
  */
 package io.camunda.webapps.schema.entities.usermanagement;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
 public record EntityJoinRelation(String name, Long parent) {
 
+  @JsonIgnore
+  public String getRouting() {
+    return parent != null ? parent.toString() : null;
+  }
+
   public static class EntityJoinRelationFactory {
 
-    private final IdentityJoinRelationshipType parent;
-    private final IdentityJoinRelationshipType child;
+    private final String parentName;
+    private final String childName;
 
-    public EntityJoinRelationFactory(
-        final IdentityJoinRelationshipType parent, final IdentityJoinRelationshipType child) {
-      this.parent = parent;
-      this.child = child;
+    public EntityJoinRelationFactory(final String parentName, final String childName) {
+      this.parentName = parentName;
+      this.childName = childName;
     }
 
     public EntityJoinRelation createParent() {
-      return new EntityJoinRelation(parent.getType(), null);
+      return new EntityJoinRelation(parentName, null);
     }
 
     public EntityJoinRelation createChild(final long parentKey) {
-      return new EntityJoinRelation(child.getType(), parentKey);
+      return new EntityJoinRelation(childName, parentKey);
     }
-  }
-
-  public enum IdentityJoinRelationshipType {
-    GROUP("group"),
-    MEMBER("member");
 
-    private final String type;
+    public String getChildName() {
+      return childName;
+    }
 
-    IdentityJoinRelationshipType(final String type) {
-      this.type = type;
+    public String getChildKey(final long parentKey, final long childKey) {
+      return "%s-%s/%s-%s".formatted(parentName, childName, parentKey, childKey);
```

Comment:
The reason for this change is that ES supports multiple relationships and we might make use of that when an entity has multiple many-to-many relations, e.g. a group could have a collection of members and a collection of admins.
I agree we should use the same pattern for all m2m relations that use EntityJoinRelation. I think I forgot to update groups.

I don't think the space usage matters. It's going to be less than a hundred bytes per document and don't think we'll have more than a million child documents.

Reply:
Where would we expose the m2m relation keys to users?

Reply:
> Depending on how we use these keys, it might be more difficult to parse it with this format.

I believe these ids should never be parsed. They are opaque identifiers for documents in ES/OS.

Reply:
> Where would we expose the m2m relation keys to users?

💭 Self-managed installations have access to their ES/OS instances and users/customers can use this data as they like. They can decide to parse these keys, and do many other interesting things with it 😄. However, I can agree that this is more hypothetical. 

However, having a good UX, a consistent codebase and key patterns is a direct benefit both to Camunda engineers and users.

> The reason for this change is that ES supports multiple relationships and we might make use of that when an entity has multiple many-to-many relations, e.g. a group could have a collection of members and a collection of admins.

👍 I really like this since it makes this code more future-proof, i.e. we can very easily expand in the direction you're suggesting. However, this change belongs to the Exporter domain, so I feel that we shouldn't make the decision that easily, and align with Ben and maybe the Exporter team. This is another reason why I would prefer to extract the refactorings in a separate PR, which wouldn't block progress on the re-architecture epic. We can take more time to discuss and validate this decision there.

Reply:
I feel I am missing some context as I've not been close to the core implementation of the parent-child relation, but in general we should align with the other keys of a similar purpose. I would suggest that if we do change it then it would be as @koevskinikola suggested (in a targetted broad PR).

Reply:
There are exactly two instances where we're using the generic EntityJoinRelation mechanism right now: role and group members. Since it's supposed to work for arbitrary many-to-many relationships, it should, in my opinion, handle the case of multiple relations between entities. Making that work requires that we change the key scheme a bit. It's cheap to do that now as there's no data to migrate.
I should have extracted that change into a separate commit, but I didn't expect it to be controversial. It's a minor bugfix.